controller/volume_controller.go

```diff
@@ -2549,7 +2551,9 @@ func (vc *VolumeController) isResponsibleFor(v *longhorn.Volume, defaultEngineIm
 		return false, err
 	}
 
-	return (isResponsible && currentNodeEngineAvailable) || (!preferredOwnerEngineAvailable && !currentOwnerEngineAvailable && currentNodeEngineAvailable), nil
+	return (currentNodeEngineAvailable && isResponsible) ||
```

Comment:
You might want to use variables like this, similar to the ControllerIsResponsibleFor, might make it easier to digest what's going on :)

```
	isPreferredOwner := currentNodeEngineAvailable && isResponsible
	continueToBeOwner := currentNodeEngineAvailable && !preferredOwnerEngineAvailable && vc.controllerID == v.Status.OwnerID
	requiresNewOwner := currentNodeEngineAvailable && !preferredOwnerEngineAvailable && !currentOwnerEngineAvailable
	return isPreferredOwner || continueToBeOwner || requiresNewOwner
```

Reply:
Thank you. Fixed